### ISSUE TITLE
torchvision CPU nightly wheels are stored in the wrong index

### ISSUE THREAD

**Issue by pmeier** [pytorch/vision#6103]
The nightlies are in https://download.pytorch.org/whl/nightly/cpu/torchvision/torchvision rather than https://download.pytorch.org/whl/nightly/cpu/torchvision/. 

Installing with

```shell
pip install --pre torchvision --extra-index-url https://download.pytorch.org/whl/nightly/cpu/
```

will pull the stable wheel from PyPI.

cc @malfet @seemethere

**Comment by tgaddair** [comment]
Is this related to https://github.com/pytorch/vision/issues/6099?

Some time today we were no longer able to install nightly `torch` with `torchvision`. `torchvision` will instead force downloading the stable torch package:

```
(base) ray@46b3d93b9717:/ludwig$ pip install --pre torch torchvision --extra-index-url https://download.pytorch.org/whl/nightly/cpu
Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/nightly/cpu
Collecting torch
  Downloading https://download.pytorch.org/whl/nightly/cpu/torch-1.13.0.dev20220527%2Bcpu-cp38-cp38-linux_x86_64.whl (189.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 189.3/189.3 MB 7.5 MB/s eta 0:00:00
Collecting torchvision
  Downloading torchvision-0.12.0-cp38-cp38-manylinux1_x86_64.whl (21.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 21.0/21.0 MB 16.9 MB/s eta 0:00:00
Requirement already satisfied: typing-extensions in /home/ray/anaconda3/lib/python3.8/site-packages (from torch) (4.2.0)
Collecting torch
  Downloading torch-1.11.0-cp38-cp38-manylinux1_x86_64.whl (750.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 750.6/750.6 MB 910.5 kB/s eta 0:00:00
```

For now our workaround is to pin to a specific torchvision wheel by its URL, but this obviously won't be sustainable as we would need to keep manually updating it for each nightly. So would be great to get this fixed soon, I suspect other projects testing against nightly torch are similarly blocked.

**Comment by pmeier** [comment]
> Is this related to https://github.com/pytorch/vision/issues/6099?

Nope. The cause of #6099 is a wrong installation command. Your command is fine, but our index is broken.

In the mean time, you can also install with 

```shell
pip install --pre torch torchvision -f https://download.pytorch.org/whl/nightly/torch_nightly.html
```

This is fine for your use case, but be aware that you can only install PyTorch wheels with this, whereas the `--extra-index-url` can also pull from PyPI if you require anything else.

**Comment by malfet** [comment]
Fixed it by deleting spurious /whl/cpu/torchvision/ packages, but I'm still not sure why it caused this weird nesting problem

**Comment by tgaddair** [comment]
Thanks for the workaround @pmeier.

@malfet, thanks for looking into this, though I am still unable to use `--extra-index-url` at this time.

**Comment by datumbox** [comment]
I also can confirm it worked on my systems. Perhaps it takes a bit of time until the change propagates in all regions?

**Comment by tgaddair** [comment]
Thanks @malfet @datumbox, looks like it's actually coming from `torchtext` now (we install it together with `torch` and `torchvision`). I'll follow-up with them, likely it's a similar issue as the one here.  Thanks for the quick fix!

**Comment by malfet** [comment]
@tgaddair hmm, can you please elaborate, what problem are you seeing with torchtext? (As I were able to install the full pack of vision, audio and text on my linux devbox)

**Comment by tgaddair** [comment]
Hey @malfet, I spoke too soon, that turned out to be my issue (already had torchtext in the env). I haven't put together a good repo for what I'm seeing in CI locally yet, but you can see the problem here when trying to install torchvision nightly with python 3.9:

https://github.com/ludwig-ai/ludwig/runs/6657822665?check_suite_focus=true#step:7:79

I'll also try re-running to see if it was a replication issue as @datumbox suggested.

**Comment by tgaddair** [comment]
Okay, seems to be working again, so I'm guessing it was as @datumbox suggested. Thanks everyone! Apologies for the back and forth.

**Comment by datumbox** [comment]
@tgaddair  Thanks for confirming. 

@malfet Thanks for the help fixing the issue.

**Comment by tgaddair** [comment]
Hey @datumbox @malfet, looks like this issue has resurfaced again today:

https://github.com/ludwig-ai/ludwig/runs/6794939594?check_suite_focus=true#step:7:78

**Comment by tgaddair** [comment]
This issue is still popping up from time to time. We tried working around it using @pmeier's suggestion to change:

```
pip install --pre torch torchtext torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/torch_nightly.html       
```

to:

```
pip install --pre torch torchtext torchvision torchaudio -f https://download.pytorch.org/whl/nightly/torch_nightly.html
```

However, this results in installing a ROCM enabled version of torch that fails at runtime and is incompatible with some of our other dependencies.

```
torch                     1.13.0.dev20220614+rocm5.0
torchaudio                0.13.0.dev20220614+rocm5.1.1
torchtext                 0.14.0.dev20220614
torchvision               0.14.0.dev20220614+rocm5.0
```

**Comment by tgaddair** [comment]
Quick update on this, we managed to find a workaround by removing `torch` and `torchvision` from the project requirements after installing the nightlies. For whatever reason having those packages in the requirements resulted in the nightly version being overwritten and reverting back to stable. So we are unblocked on this for now.

**Comment by malfet** [comment]
@tgaddair do you mind opening a separate issue on problems you continue to experience, as it does not seem to be related to the original issue reported here.

Removed torchvision/torchvision/index.html from s3, so the the folder should no longer exist:
```
aws s3 ls s3://pytorch/whl/nightly/cpu/torchvision/torchvision
```
